### ISSUE TITLE
Hang in Heroku/Dokku push-deploy

### ISSUE THREAD

**Issue by Epigene** [owen2345/camaleon-cms#321]
Hello!

**tl;dr**
Asset precompile hangs during deployment via dokku (think private Heroku).

**long version**
When I git-push my camaleon app to a dokku-configured remote that builds an application container very much like Heroku does it, repo sync and bundle seem to work fine, but when asset precompilation initiates, I see weird calls do DB and the process seems stuck for at least 30min. Eventually it finishes, but something is clearly wrong.

I am not sure whether this is a camaleon or dokku issue, so I will post at both projects and see if I get any help. At the least, perhaps someone has tried pushing a camaleon app to Heroku and experienced something similar.

Here is a sample console output during deploy, interestingly, in precompile phase we can see camaleon-related calls to DB. Normally, for apps of similar size, precompile takes a minute or so. Ran locally (`rake assets:precompile RAILS_ENV=production`), I experience no hang.

```
Counting objects: 5, done.
Delta compression using up to 4 threads.
Compressing objects: 100% (5/5), done.
Writing objects: 100% (5/5), 976 bytes | 0 bytes/s, done.
Total 5 (delta 3), reused 0 (delta 0)
-----> Cleaning up...
-----> Building uz-veselibu.lv from herokuish...
-----> Adding BUILD_ENV to build environment...
-----> Ruby app detected
-----> Compiling Ruby/Rails
-----> Using Ruby version: ruby-2.2.3
-----> Installing dependencies using bundler 1.9.7
       Running: bundle install --without development:test --path vendor/bundle --binstubs vendor/bundle/bin -j4 --deployment
       Using i18n 0.7.0
       Using rake 10.5.0
       Using CFPropertyList 2.3.2
       Using json 1.8.3
       Using minitest 5.8.3
       Using thread_safe 0.3.5
       Using builder 3.2.2
       Using erubis 2.7.0
       Using mini_portile2 2.0.0
       Using rack 1.6.4
       Using mime-types 2.99
       Using arel 6.0.3
       Using jmespath 1.1.3
       Using bcrypt 3.1.10
       Using breadcrumbs_on_rails 2.3.1
       Using cancancan 1.13.1
       Using request_store 1.3.0
       Using excon 0.45.4
       Using formatador 0.2.5
       Using multi_json 1.11.2
       Using ipaddress 0.8.2
       Using xml-simple 1.1.5
       Using inflecto 0.0.2
       Using trollop 2.1.2
       Using mini_magick 4.3.6
       Using will_paginate 3.1.0
       Using bundler 1.9.7
       Using thor 0.19.1
       Using concurrent-ruby 1.0.0
       Using coffee-script-source 1.10.0
       Using execjs 2.6.0
       Using newrelic_rpm 3.14.2.312
       Using pg 0.18.4
       Using puma 2.15.3
       Using rails_serve_static_assets 0.0.4
       Using rails_stdout_logging 0.0.4
       Using sass 3.4.21
       Using tilt 2.0.2
       Using temple 0.7.6
       Using tzinfo 1.2.2
       Using fission 0.5.0
       Using rack-test 0.6.3
       Using mail 2.6.3
       Using nokogiri 1.6.7.1
       Using aws-sdk-core 2.2.11
       Using fog-core 1.35.0
       Using will_paginate-bootstrap 1.0.1
       Using coffee-script 2.4.1
       Using uglifier 2.7.2
       Using sprockets 3.5.2
       Using rails_12factor 0.0.3
       Using slim 3.0.6
       Using activesupport 4.2.5
       Using loofah 2.0.3
       Using rbvmomi 1.8.2
       Using fog-json 1.0.2
       Using fog-xml 0.1.2
       Using fog-local 0.2.1
       Using fog-vmfusion 0.1.0
       Using aws-sdk-resources 2.2.11
       Using rails-deprecated_sanitizer 1.0.3
       Using globalid 0.3.6
       Using activemodel 4.2.5
       Using jbuilder 2.4.0
       Using rails-html-sanitizer 1.0.2
       Using fog-vsphere 0.5.0
       Using fog-aliyun 0.1.0
       Using fog-brightbox 0.10.1
       Using fog-sakuracloud 1.7.5
       Using fog-serverlove 0.1.2
       Using fog-softlayer 1.0.3
       Using fog-storm_on_demand 0.1.1
       Using fog-atmos 0.1.0
       Using fog-aws 0.8.1
       Using fog-dynect 0.0.2
       Using fog-ecloud 0.3.0
       Using fog-google 0.1.0
       Using fog-powerdns 0.1.1
       Using fog-profitbricks 0.0.5
       Using fog-radosgw 0.0.5
       Using fog-riakcs 0.1.0
       Using fog-terremark 0.1.0
       Using fog-voxel 0.1.0
       Using fog-xenserver 0.2.3
       Using aws-sdk 2.2.11
       Using rails-dom-testing 1.0.7
       Using activejob 4.2.5
       Using activerecord 4.2.5
       Using protected_attributes 1.1.3
       Using fog 1.37.0
       Using actionview 4.2.5
       Using actionpack 4.2.5
       Using actionmailer 4.2.5
       Using draper 1.4.0
       Using meta-tags 2.1.0
       Using mobu 0.0.3
       Using railties 4.2.5
       Using sprockets-rails 3.0.0
       Using kaminari 0.16.3
       Using camaleon_cms 2.0.4.1
       Using rails 4.2.5
       Using coffee-rails 4.1.1
       Using jquery-rails 4.1.0
       Using sass-rails 5.0.4
       Using camaleon_google_analytic 0.0.2 from git://github.com/cyanpunk/camaleon-google-analytic-plugin.git (at master)
       Using camaleon_post_order 0.0.1 from git://github.com/owen2345/camaleon-post-order-plugin.git (at master)
       Using migration1x 0.0.1 from git://github.com/owen2345/camaleoncms_1xmigration.git (at master)
       Using turbolinks 2.5.3
       Bundle complete! 20 Gemfile dependencies, 108 gems now installed.
       Gems in the groups development and test were not installed.
       Bundled gems are installed into ./vendor/bundle.
       Bundle completed (2.60s)
       Cleaning up the bundler cache.
-----> Preparing app for Rails asset pipeline
       Running: rake assets:precompile
       CamaleonCms::Site Load (1.3ms)  SELECT  "cama_term_taxonomy".* FROM "cama_term_taxonomy" WHERE "cama_term_taxonomy"."taxonomy" = $1  ORDER BY "cama_term_taxonomy"."term_group" DESC, "cama_term_taxonomy"."id" ASC LIMIT 1  [["taxonomy", "site"]]
       CamaleonCms::Meta Load (2.4ms)  SELECT  "cama_metas".* FROM "cama_metas" WHERE "cama_metas"."objectid" = $1 AND "cama_metas"."object_class" = $2 AND "cama_metas"."key" = $3  ORDER BY "cama_metas"."id" ASC LIMIT 1  [["objectid", 1], ["object_class", "Site"], ["key", "main_settings"]]
       CamaleonCms::Site Load (0.8ms)  SELECT "cama_term_taxonomy".* FROM "cama_term_taxonomy" WHERE "cama_term_taxonomy"."taxonomy" = $1  ORDER BY "cama_term_taxonomy"."term_group" DESC, "cama_term_taxonomy"."id" ASC  [["taxonomy", "site"]]
       CamaleonCms::Meta Load (0.8ms)  SELECT  "cama_metas".* FROM "cama_metas" WHERE "cama_metas"."objectid" = $1 AND "cama_metas"."object_class" = $2 AND "cama_metas"."key" = $3  ORDER BY "cama_metas"."id" ASC LIMIT 1  [["objectid", 1], ["object_class", "Site"], ["key", "_default"]]
        (0.8ms)  SELECT "cama_term_taxonomy"."slug" FROM "cama_term_taxonomy" WHERE "cama_term_taxonomy"."taxonomy" = $1 AND "cama_term_taxonomy"."parent_id" = $2 AND "cama_term_taxonomy"."term_group" = $3  [["taxonomy", "plugin"], ["parent_id", 1], ["term_group", 1]]
       CamaleonCms::Site Load (0.8ms)  SELECT "cama_term_taxonomy".* FROM "cama_term_taxonomy" WHERE "cama_term_taxonomy"."taxonomy" = $1  ORDER BY "cama_term_taxonomy"."term_group" DESC  [["taxonomy", "site"]]
       CamaleonCms::Meta Load (1.3ms)  SELECT  "cama_metas".* FROM "cama_metas" WHERE "cama_metas"."objectid" = $1 AND "cama_metas"."object_class" = $2 AND "cama_metas"."key" = $3  ORDER BY "cama_metas"."id" ASC LIMIT 1  [["objectid", 1], ["object_class", "Site"], ["key", "_default"]]
       CamaleonCms::Meta Load (1.0ms)  SELECT  "cama_metas".* FROM "cama_metas" WHERE "cama_metas"."objectid" = $1 AND "cama_metas"."object_class" = $2 AND "cama_metas"."key" = $3  ORDER BY "cama_metas"."id" ASC LIMIT 1  [["objectid", 1], ["object_class", "Site"], ["key", "_default"]]
        (0.4ms)  BEGIN
       SQL (1.3ms)  UPDATE "cama_metas" SET "value" = $1 WHERE "cama_metas"."id" = $2  [["value", "{"_theme":"uzveselibu_theme","refresh_cache":true,"date_notified":"2016-02-01","save_intro":true,"permit_create_account":"","logo":"","icon":"","home_page":"","error_404":"","email":"","front_per_page":10,"admin_per_page":10,"comment_status":"pending","security_captcha_user_register":"","seo_description":"","keywords":"","seo_author":"","twitter_card":"","screenshot":"","mailer_enabled":0,"filesystem_type":"local"}"], ["id", 18]]
        (2.9ms)  COMMIT
```

**Comment by MrWilhelm** [comment]
Precompile your assets on local which will create files to public/assets folder. Then push them. Once Heroku sees you have precompiled assets, It won't try to precompile them.
